python/tk_desktop/desktop_engine_project_implementation.py

```diff
@@ -132,6 +132,9 @@ def _register_commands(self):
 
             self._project_comm.call("trigger_register_command", name, gui_properties, groups)
 
+        # Let the proxy know command registration is complete
+        self._project_comm.call_no_response("project_commands_finished")
```

Comment:
Playing around with this, I found that the 'call_no_response' method actually swallows the ValueError raised when an unknown function is detected [here](https://github.com/shotgunsoftware/tk-desktop/blob/39630_no_item_overlay/python/tk_desktop/rpc.py#L166), so there is no need to have a try/except clause at this point in the code.  It would need to be handled here if I was using the 'call()' method, but I don't think that's necessary at this point.  Also, I tried implementing a custom error [here](https://github.com/shotgunsoftware/tk-desktop/blob/39630_no_item_overlay/python/tk_desktop/rpc.py#L149), but keep in mind that if the unknown function error is being raised in the first place, then it's probably because the site configuration needs to be updated, and the custom exception wouldn't be there anyway ;)  Does this all sound reasonable?